app/models/concerns/hostext/search.rb

```diff
@@ -177,6 +178,20 @@ def search_by_config_group(key, operator, value)
         {:conditions => opts, :include => :hostgroup}
       end
 
+      def search_by_proxy(key, operator, value)
+        proxy_cond = sanitize_sql_for_conditions(["smart_proxies.name #{operator} ?", value_to_sql(operator, value)])
+        host_ids = Host::Managed.reorder('')
+                                .authorized(:view_hosts, Host)
+                                .eager_load(proxy_connections_tables)
+                                .joins("LEFT JOIN smart_proxies ON smart_proxies.id IN (#{proxy_connections_columns.join(',')})")
+                                .where(proxy_cond)
+                                .uniq
+                                .pluck('hosts.id')
```

Comment:
Unfortunately, `.select(:id).to_sql` doesn't work in rails with joins (https://github.com/rails/rails/issues/12423#issuecomment-63921085) so an inner query doesn't work here and we must use two queries.